csrc/multidevice/communicator.cpp

```diff
@@ -212,14 +212,17 @@ Communicator::Communicator(
 
 void Communicator::cleanup() {
   static bool cleaned_up = false;
-  if (cleaned_up) {
```

Comment:
QQ: I forgot why we didn't have this as a check earlier but rather just a warning.

Reply:
Anyway, glad to see it converted to an error instead.

Reply:
I probably should have. I didn't think about it too much because the rest of the function was safe enough to run again. However, with this PR and thus the added call to barrier, `cleanup` won't be able to run twice.